src/test/codewhisperer/service/telemetry.test.ts

```diff
@@ -0,0 +1,657 @@
+            assertSessionClean()
+            await backsapce(editor) // todo: without this, the following manual trigger will not be displayed in the test, investigate and fix it
+
+            await manualTrigger(editor, client, config)
+            await waitUntilSuggestionSeen()
+            await acceptByTab()
+
+            assertTelemetry('codewhisperer_userTriggerDecision', [
+                session1UserTriggerEvent({ codewhispererSuggestionState: 'Reject' }),
+                session2UserTriggerEvent(),
+            ])
+        })
+
+        it('reject - typeahead not matching after suggestion is shown then invoke another round and accept', async function () {
+            // no idea why this one doesn't work, the second inline suggestion will not be shown
+            assertSessionClean()
+            const editor = await openATextEditorWithText('', 'test.py')
+
+            await manualTrigger(editor, client, config)
+            await waitUntilSuggestionSeen()
+            await typing(editor, 'H')
+            // assert.strictEqual(editor.document.getText(), 'H')
+
+            const anotherEditor = await openATextEditorWithText('', 'anotherTest.py')
+            await manualTrigger(anotherEditor, client, config)
+            await waitUntilSuggestionSeen()
+            await acceptByTab()
+            // assert.strictEqual(anotherEditor.document.getText(), `Baz${os.EOL}Baz`)
+
+            assertTelemetry('codewhisperer_userTriggerDecision', [
+                session1UserTriggerEvent({ codewhispererSuggestionState: 'Reject' }),
+                session2UserTriggerEvent(),
+            ])
+        })
+    })
+
+    describe('on editor change, focus change', function () {
+        it('reject: trigger then open another editor', async function () {
+            assertSessionClean()
+            const editor = await openATextEditorWithText('', 'test.py', tempFolder, { preview: false })
+
+            await manualTrigger(editor, client, config)
+            await waitUntilSuggestionSeen()
+
+            await openATextEditorWithText('text in 2nd editor', 'another1.py', tempFolder, {
+                preview: false,
+            })
+            const anotherEditor = await openATextEditorWithText('text in 3rd editor', 'another2.py', tempFolder, {
+                preview: false,
+            })
+
+            assert.strictEqual(vscode.window.activeTextEditor, anotherEditor)
+            // assert.strictEqual(editor.document.getText(), '')
+            await assertTabCount(3)
+            assertTelemetry('codewhisperer_userTriggerDecision', [
+                session1UserTriggerEvent({ codewhispererSuggestionState: 'Reject' }),
+            ])
+        })
+
+        it('reject: trigger then close editor', async function () {
+            assertSessionClean()
+            const editor = await openATextEditorWithText('', 'test.py')
+
+            await manualTrigger(editor, client, config)
```

Comment:
nit: seems like all tests call `manualTrigger` -> `waitUntilSuggestionSeen` -> do something, this can probably be extract to another function?

Reply:
yea, it's intentional, I was thinking we might need test cases for discard later on.

Reply:
but actually let me do this for now, for discard case we can call them seprately